apps/src/codebridge/Settings/SettingsDropdown.tsx

```diff
@@ -0,0 +1,140 @@
+import {Button} from '@code-dot-org/component-library/button';
+import CloseButton from '@code-dot-org/component-library/closeButton';
+import SimpleDropdown, {
+  SimpleDropdownProps,
+} from '@code-dot-org/component-library/dropdown/simpleDropdown';
+import {Heading6} from '@code-dot-org/component-library/typography';
+import FocusTrap from 'focus-trap-react';
+import React, {useState} from 'react';
+import {createPortal} from 'react-dom';
+
+import codebridgeI18n from '@cdo/apps/codebridge/locale';
+import {DEFAULT_FONT_SIZE_KEY, FontSize} from '@cdo/apps/lab2/constants';
+import useDropdownPosition from '@cdo/apps/lab2/hooks/useDropdownPosition';
+import {setEditorFontSize} from '@cdo/apps/lab2/redux/lab2ViewRedux';
+import {SignInState} from '@cdo/apps/templates/currentUserRedux';
+import useOutsideClick from '@cdo/apps/util/hooks/useOutsideClick';
+import {useAppSelector, useAppDispatch} from '@cdo/apps/util/reduxHooks';
+import {trySetSessionStorage} from '@cdo/apps/utils';
+import commonI18n from '@cdo/locale';
+
+import {useCodebridgeContext} from '../codebridgeContext';
+
+import moduleStyles from './settings-dropdown.module.scss';
+
+// fontSizeOptions contains a list of value/text from the FontSize enum,
+// e.g., [{value: 'Tiny', text: 'Tiny'}, {value: 'Small', text: 'Small'}, ...]
+const fontSizeOptions: SimpleDropdownProps['items'] = Object.keys(FontSize)
+  .filter(key => isNaN(Number(key))) // Filters out the reverse enum keys.
+  .map(key => ({
+    value: key,
+    text: key,
+  }));
+interface SettingsDropdownProps {
+  closeDropdown: () => void;
+  buttonRef: React.RefObject<HTMLDivElement>;
+}
+
+const SettingsDropdown: React.FunctionComponent<SettingsDropdownProps> = ({
+  closeDropdown,
+  buttonRef,
+}) => {
+  const dropdownRef = useOutsideClick<HTMLDivElement>(closeDropdown);
+  const editorFontSizeKey = useAppSelector(
+    state => state.lab2View.editorFontSizeKey
+  );
+  const selectedFontSizeKey = editorFontSizeKey || DEFAULT_FONT_SIZE_KEY;
```

Comment:
do we need `|| DEFAULT_FONT_SIZE_KEY` since the redux value has a default of `Small`?

Reply:
https://github.com/code-dot-org/code-dot-org/pull/64981/files#r2025414415